include/CoolPropLib.h

```diff
@@ -290,6 +301,17 @@
      * @return 
      */
     EXPORT_CODE void CONVENTION AbstractState_set_fractions(const long handle, const double* fractions, const long N, long *errcode, char *message_buffer, const long buffer_length);
+    /**
+     * @brief Get the molar fractions for the AbstractState
+     * @param handle The integer handle for the state class stored in memory
+     * @param fractions The array of fractions
+     * @param N number of fluids
+     * @param errcode The errorcode that is returned (0 = no error, !0 = error)
+     * @param message_buffer A buffer for the error code
+     * @param buffer_length The length of the buffer for the error code
+     * @return 
+     */
+    EXPORT_CODE void CONVENTION AbstractState_get_mole_fractions(const long handle, double *& fractions, long *N, long *errcode, char *message_buffer, const long buffer_length);
```

Comment:
fractions should be pre-allocated buffer of given length passed into this function